### ISSUE TITLE
CentOS 8

### ISSUE THREAD

**Issue by emper0r** [githubixx/ansible-role-wireguard#115]
Hi... i'm trying to implement this role by default on CentOS 8 ... 
```
kernel: 4.18
centos version: 8.3.2011
```
output
```
fatal: [firewall]: FAILED! => changed=false
  attempts: 10
  failed_when_result: true
  invocation:
    module_args:
      name: wireguard
      params: ''
      state: present
  msg: |-
    modprobe: FATAL: Module wireguard not found in directory /lib/modules/4.18.0-240.1.1.el8_3.x86_64
  name: wireguard
  params: ''
  rc: 1
  state: present
  stderr: |-
    modprobe: FATAL: Module wireguard not found in directory /lib/modules/4.18.0-240.1.1.el8_3.x86_64
  stderr_lines: <omitted>
  stdout: ''
  stdout_lines: <omitted>

```

**Comment by githubixx** [comment]
Maybe you've multiple kernels around and the `wireguard` module gets installed into the wrong directory. Maybe try `find /lib/modules/ -name "wireguard.*" -ls` to see if it was installed somewhere in `/lib/modules/` path.

**Comment by emper0r** [comment]
Hi @githubixx 

i got this
```
 find /lib/modules/ -name "wireguard.*" -ls
  1000550      0 lrwxrwxrwx   1  root     root           63 Aug 16 16:30 /lib/modules/4.18.0-305.12.1.el8_4.x86_64/weak-updates/wireguard/wireguard.ko -> /lib/modules/4.18.0-305.el8.x86_64/extra/wireguard/wireguard.ko
  8412022    332 -rw-r--r--   1  root     root       338481 Jun  6 14:39 /lib/modules/4.18.0-305.el8.x86_64/extra/wireguard/wireguard.ko

```

**Comment by githubixx** [comment]
Well `modprobe` can't do that much if the `wireguard` kernel module isn't there :wink: The error says:
`Module wireguard not found in directory /lib/modules/4.18.0-240.1.1.el8_3.x86_64`

But the module is in `/lib/modules/4.18.0-305.el8.x86_64/extra/wireguard/wireguard.ko`. So looks like you upgraded the kernel. In this case the `kmod-wireguard` package needs to be re-installed. I've no idea about CentOS but Archlinux rebuilds DKMS modules automatically after a kernel update which places `wireguard.ko` in the correct directory.

Sadly that's a problem with this old kernels that don't include the Wireguard kernel module by default and must be therefore installed separately. Wiregard module was included in kernel 5.6 but other distributions like Ubuntu 20 e.g. backported it to their default 5.4 kernel.

**Comment by emper0r** [comment]
ok.. it's working removing old one and installing the new one.
